src/contract_info.py

```diff
@@ -41,6 +41,8 @@ def token_supply(web3, token_address, abi, decimals):
     )
 
     try:
+        if decimals is None:
+            decimals = contract.functions.decimals().call()
```

Comment:
Technically we can't always assume that there's a decimals function to call on an ERC20, but in this case it should be fine since the tokens we are tracking all have the function.

Reply:
Yeah that's why I made it optional

Either we can specify it manually, or if not specified manually we try calling from the `decimals` function